Angular Modules Grid sample/src/app/master-view/master-view.component.ts

```diff
@@ -0,0 +1,196 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { IGridRowEventArgs, IgxDialogComponent, IgxGridComponent } from '@infragistics/igniteui-angular';
+import { AddressForm, CustomersForm, CustomersType, FormType } from '../models/northwind/customers-type';
+//import { Address, Customer } from '../models/northwind/northwind-forms/northwind-forms';
+import { NorthwindService } from '../services/northwind.service';
+
+@Component({
+  selector: 'app-master-view',
+  templateUrl: './master-view.component.html',
+  styleUrls: ['./master-view.component.scss']
+})
+export class MasterViewComponent implements OnInit {
+  @ViewChild('grid', { read: IgxGridComponent, static: true }) public grid: IgxGridComponent;
+  @ViewChild('form', { read: IgxDialogComponent, static: true }) public dialog: IgxDialogComponent;
+  public northwindCustomers: CustomersType[] = [];
+  public customer!: FormGroup<FormType<CustomersType>>;
+  public errorMessage: string = '';
+  public dialogTitle: string = '';
+  public confirmText: string = '';
+  public confirmEvent: string = '';
+  public isDeleteButtonHidden: boolean = false;
+
+  constructor(private northwindService: NorthwindService) {
+    this.customer = new FormGroup<FormType<CustomersType>>({
+      customerID: new FormControl(''),
+      companyName: new FormControl('', Validators.required),
+      contactName: new FormControl('', Validators.pattern("^[a-zA-Z]+( [a-zA-Z]+)*$")),
+      contactTitle: new FormControl('', Validators.required),
+      address: new FormGroup<AddressForm>({
```

Comment:
We should consider if it is best to generate all the properties (even if not all of them are defined in the application) or to type the object which would allow to not generate "redundant" properties. The root of this dilemma is FormGroup not allowing Partial<Type>